scripts/utils/PromisePool.ts

```diff
@@ -0,0 +1,34 @@
+class PromisePool<T> {
+    /**
+     * The maximum number of concurrent async operations.
+     */
+    private readonly concurrency: number;
+
+    /**
+     * The set of currently-executing async operations.
+     */
+    private executing = new Set<Promise<T>>();
+
+    constructor({concurrency}: {concurrency: number} = {concurrency: 8}) {
```

Comment:
NAB: How about
```suggestion
    constructor(concurrency: number = 8) {
```
I think it would be a bit more readable in this case